external_modules/bitcoin/methods/rewindTransactions.ts

```diff
@@ -0,0 +1,107 @@
+import { Bitcoin } from '../index'
+import meta from '../../common/db/models/meta'
+import { ObjectId } from 'mongodb'
+import Account from '../../common/db/models/account'
+import Transaction from '../../common/db/models/transaction'
+import { ReceivedPending } from '../../common/db/models/pendingTx'
+import { TxReceived } from '../../common'
+
+async function formatTransaction(txInfo: any): Promise<TxReceived|void> {
+	if(txInfo.category != 'receive') return
+
+	const address: TxReceived['account'] = txInfo.address
+
+	/** Verifica se a transação é nossa */
+	const account = await Account.findOne({ account: address })
+	if (!account) return
+
+	const formattedTransaction: TxReceived = {
+		txid:          txInfo.txid,
+		status:        'pending',
+		confirmations: txInfo.confirmations,
+		account:       address,
+		amount:        txInfo.amount,
+		timestamp:     txInfo.time * 1000 // O timestamp do bitcoin é em segundos
+	}
+
+	return formattedTransaction
+}
+
+
+export async function rewindTransactions(this: Bitcoin, newBlockhash: string) {
+	this.canSincronize = false
+	let blockhash = (await meta.findOne({info: 'lastSyncBlock'}))?.details
+
+	if (!blockhash) {
+		blockhash = (await this.rpc.blockInfo(newBlockhash))?.previousblockhash
+		if (!blockhash)
+			return
+
+		await meta.updateOne({
+			info: 'lastSyncBlock'
+		}, {
+			details: blockhash
+		}, {
+			upsert: true
+		})
+	}
+	const { transactions }: any = await this.rpc.listSinceBlock(blockhash)
+
+
+	for (const transaction of transactions) {
+		try {
+			const _transaction = await formatTransaction(transaction)
+			if (!_transaction) return
+			console.log( 'received transaction', _transaction ) //remove
+
+			/** Salva a nova transação no database */
+			await new Transaction({
+				txid : _transaction.txid,
+				type: 'receive',
+				account: _transaction.account
+			}).save().catch()
```

Comment:
Esse catch está sem um handler

Reply:
alguma ideia de que tipo de handler eu colocaria nisso ? se der duplicate key, nao é um erro que deveria ser tratado

Reply:
Faz ele ignorar o duplicate key e se for outro dá erro msm, pq daí teve mesmo alguma coisa errada